src/Compilers/VisualBasic/Portable/Parser/Parser.vb

```diff
@@ -4615,21 +4615,21 @@ checkNullable:
                 value = ParseExpressionCore()
 
             ElseIf modifiers.Any AndAlso modifiers.Any(SyntaxKind.OptionalKeyword) Then
-
-                equals = ReportSyntaxError(InternalSyntaxFactory.MissingPunctuation(SyntaxKind.EqualsToken), ERRID.ERR_ObsoleteOptionalWithoutValue)
-                value = ParseExpressionCore()
-
+                If CheckFeatures(Feature.ImplicitDefaultValueOnOptionalParameter, _scanner.Options) = False Then
+                    equals = ReportSyntaxError(InternalSyntaxFactory.MissingPunctuation(SyntaxKind.EqualsToken), ERRID.ERR_ObsoleteOptionalWithoutValue)
+                    value = ParseExpressionCore()
+                End If
             End If
-
             Dim initializer As EqualsValueSyntax = Nothing
 
-            If value IsNot Nothing Then
+            If (equals IsNot Nothing) AndAlso (value IsNot Nothing) Then
```

Comment:
I must be missing something obvious here. What is '''equals IsNot Nothing'' condition supposed to change?  Could you give an example where this makes an observable difference in behavior and what that difference is? Also, is there a unit-test that demonstrates the change in behavior?